eng/common/pipelines/templates/jobs/sdk-generation.yml

```diff
@@ -0,0 +1,200 @@
+parameters:
+  - name: SpecRepourl
+    type: string
+  - name: SpecRepoBranch
+    type: string
+    default: 'main'
+  - name: SpecRepoCommit
+    type: string
+    default: 'HEAD'
+  - name: SdkRepoUrl
+    type: string
+  - name: SdkRepoBranch
+    type: string
+    default: 'main'
+  - name: ConfigType
+    type: string
+    values:
+      - 'TypeSpec'
+      - 'AutoRest'
+  - name: TspConfigPath
+    type: string
+  - name: ReadmePath
+    type: string
+  - name: ApiVersion
+    type: string
+    default: 'not-specified'
+  - name: CodePushRequired
```

Comment:
@chidozieononiwu please combine `CodePushRequired` and `SdkPrRequired` parameters to a single one `SkipSdkPullRequestCreation` as discussed.